uniflow/op/model/lm/model_server.py

```diff
@@ -262,81 +264,88 @@ class HuggingfaceModelServer(AbsModelServer):
-    ) -> None:
+    def __init__(self, prompt_template, model_config: Dict[str, Any]) -> None:
         # import in class level to avoid installing transformers package
         super().__init__(prompt_template, model_config)
-        self._model_config = HuggingfaceModelConfig(**self._model_config)
-        if self._model_config.neuron is False:
-            try:
-                from transformers import (  # pylint: disable=import-outside-toplevel
-                    pipeline,
-                )
-            except ModuleNotFoundError as exc:
-                raise ModuleNotFoundError(
-                    "Please install transformers to use HuggingfaceModelServer. You can use `pip install transformers` to install it."
-                ) from exc
-            model, tokenizer = self._get_model()
-            # explicitly set batch_size for pipeline
-            # for batch inference.
-            self._pipeline = pipeline(
-                "text-generation",
-                model=model,
-                tokenizer=tokenizer,
-                device_map="auto",
-                do_sample=self._model_config.do_sample,
-                temperature=self._model_config.temperature,
-                num_beams=self._model_config.num_beams,
-                max_new_tokens=self._model_config.max_new_tokens,
-                num_return_sequences=self._model_config.num_return_sequences,
-                repetition_penalty=self._model_config.repetition_penalty,
-                eos_token_id=tokenizer.eos_token_id,
-                pad_token_id=tokenizer.pad_token_id,
-                batch_size=self._model_config.batch_size,
-            )
-        else:
-            if self._model_config.load_in_4bit or self._model_config.load_in_8bit:
-                self._model_config.load_in_4bit = False
-                self._model_config.load_in_8bit = False
-                print(
-                    "Neuron model does not support quantized models. load_in_4bit and load_in_8bit are automatically set to False."
-                )
-            from uniflow.op.model.lm.neuron_utils import (  # pylint: disable=import-outside-toplevel
-                Neuron,
-            )
+        self._model_config = HuggingfaceModelConfig(**model_config)
 
-            model, tokenizer = Neuron.get_neuron_model(
-                self._model_config.model_name, self._model_config.batch_size
-            )
-            self._pipeline = partial(
-                Neuron.neuron_infer,
-                model=model,
-                tokenizer=tokenizer,
-                max_new_tokens=self._model_config.max_new_tokens,
-                batch_size=self._model_config.batch_size,
+        if self._model_config.model_name == "google/gemma-7b-it":
+            self._initialize_gemma_model()
+        elif self._model_config.neuron:
+            self._initialize_neuron_model()
+        else:
+            self._initialize_default_model()
```

Comment:
You should handle neuron vs non-neuron first. Then, for neuron model, you need to handle all kinds of different ones such as Mistral, Gemma regarding their instruction token.